### PR TITLE
Amin week2

### PR DESCRIPTION
**Pull request by aminmahrami**
added external JSON file and "done" checkbox.

### CODE REVIEW

File: todo-app/src/TodoList.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  state = {
+    todos: initialTodoData,
+  }
+  //done checkbox event handler
+  handleDoneToggle = (todoId) => {
+    this.setState({
+      todos: this.state.todos.map(todo => {
+        if (todo.id === todoId) {
+          todo.done = !todo.done
```
Comment (joostlubach) [inline todo-app/src/TodoList.js]: In this line, you're mutating (modifying) the todo item. In React, when using setState, make sure to make your code immutable. the `map` function is an example of an immutable function. Can you modify this code to make it fully immutable?

File: todo-app/src/Todo.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                <div className="Todo-Description">
+                   <Description task={todo.task}/>
+                </div> 
+                    
+                <div className="Todo-Deadline">
+                   <Deadline time={todo.time}/>
+                </div> 
+
+                <div>
+                   is done : <input onClick={() => handleDoneToggle(todo.id)}
```
Comment (joostlubach) [inline todo-app/src/Todo.js]: Your checkbox is "uncontrolled" (do you know why?) and as such will not display a check if the item was already done in the JSON file (see https://reactjs.org/docs/uncontrolled-components.html). What would you need to change to make this checkbox controlled?

File: todo-app/src/data/todo-data.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[{
+    "id":"1",
+    "task":"Get out of bed",
+    "time":"Wed Sep 13 2017",
+    "done": true
```
Comment (joostlubach) [inline todo-app/src/data/todo-data.json]: In relation to my comment above: when loading your app, the checkbox for this to do is unchecked.

**Comment by aminmahrami** [comment]
Yes , now I am working on week 3. i will try to make the changes that you have mentioned.

**Pull request closed without merging**